encodings/runend/src/compute/mod.rs

```diff
@@ -407,6 +416,25 @@ mod test {
         );
     }
 
+    #[test]
+    fn slice_at_end() {
+        let re_array = RunEndArray::try_new(
+            vec![7_u64, 10].into_array(),
+            vec![2_u64, 3].into_array(),
+            Validity::NonNullable,
+        )
+        .unwrap();
+
+        assert_eq!(re_array.len(), 10);
+
+        let sliced_array = slice(&re_array, re_array.len(), re_array.len()).unwrap();
+        assert!(sliced_array.is_empty());
+
+        let re_slice = RunEndArray::try_from(sliced_array).unwrap();
+        assert!(re_slice.ends().is_empty());
```

Comment:
Before this change, we would slice the values/end down into 1 item arrays instead of empty ones.